Shoko.Server/Utilities/SeriesSearch.cs

```diff
@@ -1,372 +1,378 @@
+        {
+            var parallelList = list.ToList().AsParallel();
+            bool forceAscii = Languages.PreferredNamingLanguages.ContainsOnlyLatin();
+
+            List<SearchResult<T>> results = parallelList.Select(a =>
+            {
+                List<string> titles = selector(a);
+                SearchResult<T> dist = null;
+                foreach (string title in titles)
+                {
+                    if (string.IsNullOrEmpty(title)) continue;
+                    int k = Math.Max(Math.Min((int) (title.Length / 6D), (int) (query.Length / 6D)), 1);
+                    if (query.Length <= 4 || title.Length <= 4) k = 0;
+
+                    Misc.SearchInfo<T> result =
+                        Misc.DiceFuzzySearch(title, query, k, a, forceAscii);
+                    if (result.Index == -1) continue;
+                    SearchResult<T> searchGrouping = new SearchResult<T>
+                    {
+                        Distance = result.Distance,
+                        Index = result.Index,
+                        ExactMatch = result.ExactMatch,
+                        Match = title,
+                        Result = a
+                    };
+                    if (result.Distance < (dist?.Distance ?? int.MaxValue)) dist = searchGrouping;
+                }
+
+                return dist;
+            }).Where(a => a != null && a.Index != -1).ToList().OrderBy(a => a.Index).ThenBy(a => a.Distance).ToList();
+            
+            return results;
+        }
+
+        /// <summary>
+        ///     Search for series with given query in name or tag
+        /// </summary>
+        /// <param name="query">target string</param>
+        /// <param name="userID">user id</param>
+        /// <param name="limit">The number of results to return</param>
+        /// <param name="flags">The SearchFlags to determine the type of search</param>
+        /// <returns>
+        ///     <see cref="List{SearchResult}" />
+        /// </returns>
+        public static List<SearchResult<SVR_AnimeSeries>> Search(int userID, string query, int limit, SearchFlags flags, TagFilter.Filter tagFilter = TagFilter.Filter.None)
+        {
+            query = query.ToLowerInvariant();
+
+            SVR_JMMUser user = RepoFactory.JMMUser.GetByID(userID);
+            if (user == null) throw new Exception("User not found");
+
+            ParallelQuery<SVR_AnimeSeries> allSeries = 
+                RepoFactory.AnimeSeries.GetAll().AsParallel().Where(a =>
+                a?.GetAnime() != null && (a.GetAnime().GetAllTags().Count==0 || !a.GetAnime().GetAllTags().FindInEnumerable(user.GetHideCategories())));
+            
+            ParallelQuery<AniDB_Tag> allTags = RepoFactory.AniDB_Tag.GetAll().AsParallel()
+                .Where(a =>
+                {
+                    return !user.GetHideCategories().Contains(a.TagName) &&
                            !TagFilter.IsTagBlackListed(a.TagName, tagFilter);

```

Comment:
I know I'm at fault on this line, but can you **please** change the line endings for the file back to CRLF?

Reply:
It makes it easier to see what have changed and also won't override the git (blame) history for unrelated lines

Reply:
Oh, so that's why the diff is so crazy. Should I put the line endings change in a separate commit and add a . git-blame-ignore-revs or I just leave them as they are?